[python3] Build fails if ncurses is already installed

Package: python3[core]:x64-osx -> 3.10.7#2

**Host Environment**

- Host: arm64-osx
- Compiler: AppleClang 14.0.0.14000029
-    vcpkg-tool version: 2999-12-31-unknownhash-debug
    vcpkg-scripts version: 7643561c2 2022-12-08 (26 hours ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

```
-- Using cached python-cpython-v3.10.7.tar.gz.
-- Cleaning sources at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/python3/src/v3.10.7-6160741e7f.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/python-cpython-v3.10.7.tar.gz
-- Applying patch 0001-only-build-required-projects.patch
-- Applying patch 0003-use-vcpkg-zlib.patch
-- Applying patch 0004-devendor-external-dependencies.patch
-- Applying patch 0005-dont-copy-vcruntime.patch
-- Applying patch 0008-python.pc.patch
-- Applying patch 0009-bz2d.patch
-- Applying patch 0010-dont-skip-rpath.patch
-- Applying patch 0002-static-library.patch
-- Using source at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/python3/src/v3.10.7-6160741e7f.clean
-- Getting CMake variables for x64-osx-dbg
-- Getting CMake variables for x64-osx-rel
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
-- Installing x64-osx-dbg
-- Building x64-osx-rel
-- Installing x64-osx-rel
-- Installing: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/python3_x64-osx/share/python3/copyright
-- Fixing pkgconfig file: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/python3_x64-osx/lib/pkgconfig/python-3.10-embed.pc
-- Fixing pkgconfig file: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/python3_x64-osx/lib/pkgconfig/python-3.10.pc
-- Fixing pkgconfig file: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/python3_x64-osx/debug/lib/pkgconfig/python-3.10-embed.pc
-- Fixing pkgconfig file: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/python3_x64-osx/debug/lib/pkgconfig/python-3.10.pc
CMake Error at ports/python3/portfile.cmake:270 (message):
  There should be no modules with "_failed" suffix:

  

  
  /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/python3_x64-osx/lib/python3.10/lib-dynload/_curses.cpython-310-darwin_failed.so


  
  /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/python3_x64-osx/lib/python3.10/lib-dynload/_curses_panel.cpython-310-darwin_failed.so
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)



```


**Additional context**
First install 
<details><summary>vcpkg.json</summary>

```
{
  "name": "lichtsteuerung",
  "version-string": "2.0.0",
  "dependencies": [
    "ncurses"
  ]
}

```
</details>
and then 
<details><summary>vcpkg.json</summary>

```
{
  "name": "lichtsteuerung",
  "version-string": "2.0.0",
  "dependencies": [
    "ncurses",
    "python3"
  ]
}

```
</details>

--- a/ports/python3/0012-force-disable-curses.patch
+++ b/ports/python3/0012-force-disable-curses.patch
@@ -1,0 +1,12 @@
+diff --git a/Modules/Setup b/Modules/Setup
+index 87c6a15..7ad53f5 100644
+--- a/Modules/Setup
++++ b/Modules/Setup
+@@ -375,3 +375,7 @@ xxsubtype xxsubtype.c
+ #
+ #_sqlite3 _tkinter _curses pyexpat
+ #_codecs_jp _codecs_kr _codecs_tw unicodedata
++
++*disabled*
++_curses
++_curses_panel

--- a/ports/python3/portfile.cmake
+++ b/ports/python3/portfile.cmake
@@ -16,6 +16,7 @@ set(PATCHES
     0008-python.pc.patch
     0009-bz2d.patch
     0010-dont-skip-rpath.patch
+    0012-force-disable-curses.patch
 )
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")

--- a/ports/python3/vcpkg.json
+++ b/ports/python3/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "python3",
   "version": "3.10.7",
-  "port-version": 3,
+  "port-version": 4,
   "description": "The Python programming language",
   "homepage": "https://github.com/python/cpython",
   "license": "Python-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6106,7 +6106,7 @@
     },
     "python3": {
       "baseline": "3.10.7",
-      "port-version": 3
+      "port-version": 4
     },
     "qca": {
       "baseline": "2.3.4",

--- a/versions/p-/python3.json
+++ b/versions/p-/python3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "13185f7c713eeb0ec50990488e9ea2eef9a9a276",
+      "version": "3.10.7",
+      "port-version": 4
+    },
+    {
       "git-tree": "34d2d172a62b2d8da14493b4c6ad9495cf2a28cb",
       "version": "3.10.7",
       "port-version": 3

